lib/memory_profiler/helpers.rb

```diff
@@ -28,5 +29,8 @@ def lookup_class_name(klass)
       @class_name_cache[klass] ||= ((klass.is_a?(Class) && klass.name) || '<<Unknown>>').to_s
     end
 
+    def lookup_string(obj)
+      @string_cache[obj] ||= String.new << obj
```

Comment:
I guess we do not obj.dup in case someone intercepted that, not sure. we probably should comment in the reason

Reply:
We had a problem before (fixed in 4748e7f2) where `.dup` copied the allocation tracing to the new String which then cause the retained strings report to be incorrect. I wasn't able to recreate it today so possibly it only happened with a specific version of ruby. I did change from `''` to `String.new` which allow memory_profiler to run when `RUBYOPT=--enable-frozen-string-literal` is used.